src/receipt.rs

```diff
@@ -368,6 +368,14 @@ impl Receipt {
         }
     }
 
+    pub const fn is_empty(&self) -> Option<bool> {
+        if let Some(len) = self.len() {
+            Some(len == 0)
+        } else {
+            None
+        }
```

Comment:
I think a one-liner here would be more clear on what we are doing

```suggestion
        self.len().map(|l| l == 0)
```

Reply:
Tried it before but `is_empty` is const function so i can't call non const `map`.
maybe something like is better:
```rust
        match self.len() {
            Some(0) => Some(true),
            Some(_) => Some(false),
            None => None,
        }
```
wdyt?